src/main/java/chalmers/pimp/model/ModelImpl.java

```diff
@@ -266,4 +276,23 @@ public void undo() {
   public void redo() {
     commandManager.redo();
   }
+
+  @Override
+  public void setSelectedColor(IColor color) {
+    // A null instance is sent to this function from the color picker when the app is started.
+    // Quick fix to prevent bad stuff from occurring. Implement a better way of solving this issue?
+    // Or is it simply good enough?
+    if (color == null) {
+      return;
+    }
+
+    var cmd = CommandFactory.createChangeColorCommand(this, colorModel, color);
+    cmd.execute();
+    commandManager.insertCommand(cmd);
+  }
```

Comment:
Why is the null reference parsed at startup?

Reply:
I **believe** it is because when the color picker java fx / scenebuilder is created, it is technically a change of its value from not existing to existing but having the value "null" before being initialized? Will have to look more into it though.

Reply:
apparently, this was not the case. Removed the check and added a Objects.requireNonNull instead